### PR TITLE
FR-1649: Contested Draft Order Not Approved

### PR DESCRIPTION
**Pull request by HarryH96**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FR-1649

### Change description ###
Contested Draft Order Not Approved / Refusal Order

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedDraftOrderNotApprovedService.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        Map<String, Object> caseData = caseDetails.getData();
+        List<Object> refusalReasons = (List<Object>) caseData.get(CONTESTED_APPLICATION_NOT_APPROVED_REASONS_FOR_REFUSAL);
+
+        StringBuilder formattedRefusalReasons = new StringBuilder();
+        refusalReasons.forEach(reason -> {
+            if (formattedRefusalReasons.length() > 0) {
+                formattedRefusalReasons.append('\n');
+            }
+            formattedRefusalReasons.append("- ");
+            formattedRefusalReasons.append(((LinkedHashMap<String, LinkedHashMap>)reason).get("value").get("judgeNotApprovedReasons"));
```
Comment (ocarleton) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedDraftOrderNotApprovedService.java:96]: I think the cast can be reduced to (Map<String, Map>)

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedDraftOrderNotApprovedServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        void verifyAdditionalFieldsContested() {
+            Map<String, Object> data = data();
+
+            assertThat(data.get("ApplicantName"), is("Contested Applicant Name"));
+            assertThat(data.get("RespondentName"), is("Contested Respondent Name"));
+            assertThat(data.get("Court"),is("Nottingham County Court and Family Court"));
+            assertThat(data.get("JudgeDetails"), is("Her Honour Judge Contested"));
+            assertThat(data.get("ContestOrderNotApprovedRefusalReasonsFormatted"),
```
Comment (shaunganley) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedDraftOrderNotApprovedServiceTest.java:149]: can we get some tests for empty and one refusal reason also?

Comment (HarryH96) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedDraftOrderNotApprovedServiceTest.java:149]: Ill haggle you down on this one, edge cases are:
- no refusal reasons
- multiple refusal reasons
The test case for 1 refusal reason is a bit redundant, because if it works for multiple it works for one!
I can add a test case for empty refusal reasons
